src/SFA.DAS.Assessor.Functions/Domain/Learners/EnqueueLearnerInfoCommand.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using SFA.DAS.Assessor.Functions.Data;
+using SFA.DAS.Assessor.Functions.Domain.Learners.Interfaces;
+using SFA.DAS.Assessor.Functions.Domain.Learners.Types;
+using SFA.DAS.Assessor.Functions.ExternalApis.Approvals.OuterApi;
+
+namespace SFA.DAS.Assessor.Functions.Domain.Learners
+{
+    public class EnqueueLearnerInfoCommand : IEnqueueLearnerInfoCommand
+    {
+        public ICollector<string> StorageQueue { get; set; }
+
+        private readonly IOuterApiClient _outerApiClient;
+        private readonly ILogger<EnqueueLearnerInfoCommand> _logger;
+        private readonly IAssessorServiceRepository _assessorServiceRepository;
+
+        public EnqueueLearnerInfoCommand(IOuterApiClient outerApiClient,
+            ILogger<EnqueueLearnerInfoCommand> logger,
+            IAssessorServiceRepository assessorServiceRepository)
+        {
+            _outerApiClient = outerApiClient;
+            _logger = logger;
+            _assessorServiceRepository = assessorServiceRepository;
+        }
+
+        public async Task Execute()
+        {
+            try
+            {
+                _logger.LogInformation("EnqueueLearnerInfoCommand started");
+
+                var learnersToProcessUln = await _assessorServiceRepository.GetLearnersWithoutEmployerInfo();
+
+                if (learnersToProcessUln == null || !learnersToProcessUln.Any())
+                {
+                    _logger.LogInformation($"There is no learner to process EnqueueLearnerInfoCommand completed");
+                    return;
+                }
+
+                _logger.LogInformation($"Learners to process {learnersToProcessUln.Count}");
+
+                //1. Get all Learners From Approvals in batches
+                DateTime? extractStartTime = new DateTime(2000, 1, 1);
+                const int batchSize = 1000;
+                int batchNumber = 0;
+                int count = 0;
+                GetAllLearnersResponse learnersBatch = null;
+
+                do
+                {
+                    batchNumber++;
+                    try
+                    {
+                        learnersBatch = await _outerApiClient.Get<GetAllLearnersResponse>(new GetAllLearnersRequest(extractStartTime, batchNumber, batchSize));
+                        if (learnersBatch?.Learners == null)
+                        {
+                            _logger.LogWarning($"Failed to get learners batch: sinceTime={extractStartTime?.ToString("o", System.Globalization.CultureInfo.InvariantCulture)} batchNumber={batchNumber} batchSize={batchSize}");
```

Comment:
should we break here learnerBatch or Learners = null?

Reply:
Yes changed